core/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -660,7 +668,7 @@ public void process(MetaData metaData, @Nullable MappingMetaData mappingMd, bool
     @Override
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
-        type = in.readString();
+        type = in.readOptionalString();
```

Comment:
breaks compatibility with old clients. In case of backport, we should be careful here.